pkg/plugins/layout.go

```diff
@@ -319,58 +323,73 @@ func (dev *Disk) AddPartition(label string, size uint, fileSystem string, pLabel
 		return "", err
 	}
 
-	var pDev string
 
-	for tries := 0; tries <= 5; tries++ {
-		err = dev.ReloadPartitionTable(console)
-		if err != nil {
-			log.Errorf("Failed on reloading the partition table: %v\n", err)
-			return "", err
-		}
-		pDev, err = dev.FindPartitionDevice(part.Number, console)
-		// exit if by the fifth time we cannot get the partition device
-		if err != nil && tries == 4 {
-			return "", err
-		}
-		// If no error, we got the partition, exit the loop
-		if err == nil {
-			break
-		}
-		log.Warningf("Failed to reload %s, retrying", dev.Device)
-		time.Sleep(1 * time.Second)
+	err = dev.ReloadPartitionTable(console)
+	if err != nil {
+		log.Errorf("Failed on reloading the partition table: %v\n", err)
+		return "", err
+	}
+
+	pDev, err := dev.FindPartitionDevice(part.Number, console)
+	if err != nil {
+		return "", err
 	}
 
 	mkfs := MkfsCall{part: part, customOpts: []string{}, dev: pDev}
 	return mkfs.Apply(console)
 }
 
 func (dev Disk) ReloadPartitionTable(console Console) error {
-	out, err := console.Run(fmt.Sprintf("blockdev --rereadpt %s", dev))
-	if err != nil {
-		return errors.New(fmt.Sprintf("Could not reload partition table: %s", out))
-	}
-	out, err = console.Run("sync")
-	if err != nil {
-		return errors.New(fmt.Sprintf("Could not sync: %s", out))
+	for tries := 0; tries <= partitionTries; tries++ {
+		log.Debugf("Trying to reread the partition table of %s (try number %d)", dev, tries+1)
+
+		out, err1 := console.Run(fmt.Sprintf("partprobe %s", dev))
+		if err1 != nil && tries == (partitionTries - 1) {
+			return errors.New(fmt.Sprintf("Could not reload partition table: %s", out))
+		}
+
+		// Extra: Try also to use blockdev to re-read the partition.
+		// This might fail during early boot so ignore any errors, just run it to try to reread the partition anyway
+		_, _ = console.Run(fmt.Sprintf("blockdev --rereadpt %s", dev))
```

Comment:
urks... 😅  Is there any reason it might fail on early boot?

Reply:
Just keeps saying that cannot access the device, probably the loop device that is mounted for recovery??? :/

Partprobe works perfectly in there, no issues, re-reads everything.

Reply:
I dropped this, we can always add it afterwards if we see that the waiting is not enough (should be enough, otherwise we have the disk in use and that would be bad)